src/Handler/AliasHandler.php

```diff
@@ -39,26 +40,44 @@ public function __construct(ObjectManager $manager, AliasCreator $creator)
 
     /**
      * @param array $aliases
+     * @param int   $limit
      *
      * @return bool
      */
-    public function checkAliasLimit(array $aliases): bool
+    public function checkAliasLimit(array $aliases, int $limit = self::ALIAS_LIMIT_CUSTOM): bool
     {
-        return (count($aliases) < self::ALIAS_LIMIT) ? true : false;
+        return (count($aliases) < $limit) ? true : false;
     }
 
     /**
-     * @param User        $user
-     * @param null|string $localPart
+     * @param User $user
      *
      * @return Alias|null
      *
      * @throws ValidationException
      */
-    public function create(User $user, ?string $localPart): ?Alias
+    public function createRandom(User $user): ?Alias
```

Comment:
Why not keep a single method `create()` and use the condition `if (null === $localPart)` to distinguish between randomAlias and customAlias?

```
$limit = (null === $localPart) ? self::ALIAS_LIMIT_RANDOM : self::ALIAS_LIMIT_CUSTOM;
```

The `repository->find*ByUser()` condition needs to be unified anyway (see next two comments).